src/precompile.jl

```diff
@@ -1,3 +1,14 @@
+macro warnpcfail(ex::Expr)
+    modl = __module__
+    file = __source__.file === nothing ? "?" : String(__source__.file)
+    line = __source__.line
+    quote
+        $(esc(ex)) || @warn """precompile directive
```

Comment:
Sorry for the late feedback. Maybe a `@debug` instead of `@warn` can be more helpful when someone wants to investigate whether the precompilations are still valid?

Reply:
Did you mean https://github.com/JuliaLang/julia/pull/41447?

Reply:
Exactly! And doing it upstream is better than doing it here.